pkg/syncer/resource_syncer.go

```diff
@@ -158,6 +158,9 @@ type ResourceSyncerConfig struct {
 
 	// MetricOpts used to pass name and help text to resource syncer Gauge
 	MetricOpts *prometheus.GaugeOpts
+
+	// When used from broker.syncer, SyncCounter will be shared by several resourceSyncer
+	SyncCounter *prometheus.GaugeVec
```

Comment:
```suggestion
	// MetricGauge if specified, used to record counter metrics.
	MetricGauge *prometheus.GaugeVec
```

Reply:
I requested the `SyncCounter` name; `MetricGauge` doesn’t contain any useful information IMO — we know it’s a metric because it’s a Prometheus object, and we know it’s a gauge because it’s a `GaugeVec`, and featuring both of those in the field name ties it to its type. I’d rather the name reflect the use.

Reply:
ok - makes sense. Perhaps we should rename `MetricOpts` to `SyncCounterOpts`.  The main reason I suggested `MetricGauge` was to align it with `MetricOpts`.

Reply:
@skitt - Right, I forgot about that..